local-cli/runAndroid/runAndroid.js

```diff
@@ -12,91 +12,64 @@ const chalk = require('chalk');
-  const args = parseCommandLine([{
-    command: 'install-debug',
-    type: 'string',
-    required: false,
-  }, {
-    command: 'root',
-    type: 'string',
-    description: 'Override the root directory for the android build (which contains the android directory)',
-  }, {
-    command: 'flavor',
-    type: 'string',
-    required: false,
-  }, {
-    command: 'variant',
-    type: 'string',
-    required: false,
-  }], argv);
-
-  args.root = args.root || '';
-
-  if (!checkAndroid(args)) {
+function runAndroid(argv, config, args) {
+  if (!checkAndroid(args.root)) {
     console.log(chalk.red('Android project not found. Maybe run react-native android first?'));
     return;
   }
 
-  resolve(isPackagerRunning().then(result => {
+  return isPackagerRunning().then(result => {
     if (result === 'running') {
-      console.log(chalk.bold(`JS server already running.`));
+      console.log(chalk.bold('JS server already running.'));
     } else if (result === 'unrecognized') {
-      console.warn(chalk.yellow(`JS server not recognized, continuing with build...`));
+      console.warn(chalk.yellow('JS server not recognized, continuing with build...'));
     } else {
       // result == 'not_running'
-      console.log(chalk.bold(`Starting JS server...`));
+      console.log(chalk.bold('Starting JS server...'));
       startServerInNewWindow();
     }
-    buildAndRun(args, reject);
-  }));
-}
-
-// Verifies this is an Android project
-function checkAndroid(args) {
-  return fs.existsSync(path.join(args.root, 'android/gradlew'));
+    return buildAndRun(args);
+  });
 }
 
 // Builds the app and runs it on a connected emulator / device.
-function buildAndRun(args, reject) {
+function buildAndRun(args) {
   process.chdir(path.join(args.root, 'android'));
   try {
     const cmd = process.platform.startsWith('win')
       ? 'gradlew.bat'
       : './gradlew';
 
     const gradleArgs = [];
-    if (args['variant']) {
-        gradleArgs.push('install' +
-          args['variant'][0].toUpperCase() + args['variant'].slice(1)
-        );
-    } else if (args['flavor']) {
-        console.warn(chalk.yellow(
-          `--flavor has been deprecated. Use --variant instead`
-        ));
-        gradleArgs.push('install' +
-          args['flavor'][0].toUpperCase() + args['flavor'].slice(1)
-        );
+    if (args.variant) {
+      gradleArgs.push('install' +
+        args.variant[0].toUpperCase() + args.variant.slice(1)
+      );
+    } else if (args.flavor) {
+      console.warn(chalk.yellow(
+        '--flavor has been deprecated. Use --variant instead'
+      ));
+      gradleArgs.push('install' +
+        args.flavor[0].toUpperCase() + args.flavor.slice(1)
+      );
     } else {
-        gradleArgs.push('installDebug');
+      gradleArgs.push('installDebug');
     }
 
-    if (args['install-debug']) {
-      gradleArgs.push(args['install-debug']);
+    if (args.installDebug) {
+      gradleArgs.push(args.installDebug);
```

Comment:
@mkonicek btw isn't that a bug? Why do we push `installDebug` value here? Wasn't that meant to be an extra arguments only when `installDebug` is used? (as defined on line 68?) Otherwise we are passing installDebug option to another variant.